File: public/docs/_examples/intro/ts/src/app/hero-detail.component.ts
import {Component, Input} from 'angular2/angular2';
import {Hero} from './hero';

@Component({
  selector: 'hero-detail',
  templateUrl: 'app/hero-detail.component.html',
  directives: [HeroDetailComponent]
})
export class HeroDetailComponent {
  @Input() hero:Hero;
}

File: public/docs/_examples/intro/dart/lib/hero_detail_component.dart
library developer_guide_intro.hero_detail_component;

import 'package:angular2/angular2.dart';
import 'package:developer_guide_intro/hero.dart';

@Component(selector: 'hero-detail', templateUrl: 'hero_detail_component.html')
class HeroDetailComponent {
  @Input()
  Type hero = Hero;
}
